### PR TITLE
add current temperature

### PR DESCRIPTION
**Pull request by mshulman**
Write current temperature to "environment.outside.temperature"

### CODE REVIEW

File: openweather.js [hunk 1]
```diff
@@ -131,6 +133,16 @@ function onPositionUpdate(value, callback) {
 
         owm.setCoordinate(value.latitude, value.longitude);
         log("OWM Coordinates "+value.latitude+","+value.longitude);
+
+        // get the Temperature  
+	    owm.getTemperature(function(err, temp){
+		    if (!err)
+            {
+                console.log('Current temperature:'+ temp);
```
Comment (inspired-technologies) [inline openweather.js]: plugin should not write directly to the console - use log function instead.

File: openweather.js [hunk 2]
```diff
@@ -131,6 +133,16 @@ function onPositionUpdate(value, callback) {
 
         owm.setCoordinate(value.latitude, value.longitude);
         log("OWM Coordinates "+value.latitude+","+value.longitude);
+
+        // get the Temperature  
+	    owm.getTemperature(function(err, temp){
```
Comment (inspired-technologies) [inline openweather.js:138]: This will return exactly the same data as getSmartJSON further down below

File: openweather.js [hunk 3]
```diff
@@ -28,6 +28,7 @@ const subscriptions = [
     rain: 4,
     weathercode : 200 
 */
+const currentTempPath = "environment.outside.temperature";
```
Comment (inspired-technologies) [inline openweather.js:31]: This path may conflict with data exposed by other plugins like BME280 or RUUVI tag

File: openweather.js [hunk 4]
```diff
@@ -273,6 +287,7 @@ function prepareUpdate(forecast, weather, full) {
             buildDeltaUpdate(simpleWeatherCode, weather.weathercode)
         ];
         case 'meta-simple': return [
+            buildDeltaUpdate(currentTempPath, latest.currentTemp),
```
Comment (inspired-technologies) [inline openweather.js]: Value is wrong for meta-data, needs hold units, ie. `{ units: "K"} `

**Review by inspired-technologies — CHANGES_REQUESTED** [review]
Thx for raising this PR; however, there are a number of concerns with this requets:

1. The path proposed may easily conflict with other plugins like the bme280 or RUUVI which measure temperature, pressure and humidity directly on board. Any of these would have clear preference over projected data from an API.
2. `owm.getTemperature` would not give any different value than already collected in  simple.temp ejected into `environment.forecast.temperature`. It would also be the same value as collected in full if offset is set to 0. 
3. Value needs to be set in Kelvin (as required by SignalK spec) and the meta-data value needs to be set appropriately.

File: openweather.js (hunk 3 above)
Comment (mshulman) [reply openweather.js:31]: The SignalK server can resolve multiple paths from different senders.  This is preferable to having different paths for the same value.

For example, multiple GPS units on the network may all provide lat/lon data.

File: openweather.js (hunk 2 above)
Comment (mshulman) [reply openweather.js:138]: I didn't realize. will fix.

File: openweather.js (hunk 1 above)
Comment (mshulman) [reply openweather.js]: fixed

File: openweather.js (hunk 4 above)
Comment (mshulman) [reply openweather.js]: fixed, thank you

File: openweather.js (hunk 2 above)
Comment (mshulman) [reply openweather.js:138]: fixed.

File: openweather.js (hunk 3 above)
Comment (inspired-technologies) [reply openweather.js:31]: Understood and acknowledged - however, this is still a different situation to capturing gps with multiple devices onboard, ie. local sensors capturing significantly different data than returned from the API which can only be an approxmiate median value (if any when in an off-shore location). Anyone following that data (eg. by uploading to a timeseries on influx or for feeding into predictions) will then have to differentiate per data source if the data is trustworthy. Hence, the additional path would at least have to be configurable. In addition need to make clear whether or not to treat `pressure` and `humidity` in the same way.

**Comment by mshulman** [comment]
I think I've made all the requested changes. Is there anything left?

**Comment by inspired-technologies** [comment]
We've been re-factoring the plugin code a little bit, but indeed - there were 2 questions open to be clarified: (a) why not handling pressure and humidity in the same way as proposed for temperature and (b) make the path(s) additions configurable via options to leave it to the user and avoid potential data conflicts. 

There was a similar ask in another plugin which is in testing right now - hence, it may be best to have this feature implemented consistently,

**Comment by inspired-technologies** [comment]
Closing this PR as re-solved by published [v0.7](https://www.npmjs.com/package/openweather-signalk/v/0.7.0)

**Pull request closed without merging**